testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/client/ClientPoliciesTest.java

```diff
@@ -1279,6 +1281,80 @@ public void testHolderOfKeyEnforceExecutor() throws Exception {
+        });
+        registerCondition("AnyClientConditionFactory", policyName);
+        logger.info("... Registered Condition : AnyClientConditionFactory");
+
+        createExecutor(executorName, TestRaiseExeptionExecutorFactory.PROVIDER_ID, null, (ComponentRepresentation provider) -> {
+            provider.getConfig().put(TestRaiseExeptionExecutorFactory.TARGET_CP_EVENTS, new ArrayList<>(Arrays.asList(ClientPolicyEvent.REGISTERED.toString())));
+        });
+        registerExecutor(executorName, policyName);
+        logger.info("... Registered Executor : " + executorName);
+
+        String clientId = null;
+        try {
+            try {
+                createClientByAdmin(clientName, (ClientRepresentation clientRep) -> {
+                });
+                fail();
+            } catch (ClientPolicyException cpe) {
+                assertEquals(ClientPolicyEvent.REGISTERED.toString(), cpe.getError());
+            }
+
+            updateExecutor(executorName, (ComponentRepresentation provider) -> {
+                provider.getConfig().put(TestRaiseExeptionExecutorFactory.TARGET_CP_EVENTS, new ArrayList<>(Arrays.asList(ClientPolicyEvent.UPDATED.toString())));
+            });
+
+            clientId = getClientByAdminWithName(clientName).getId();
+            assertEquals(true, getClientByAdmin(clientId).isEnabled());
+            try {
+                updateClientByAdmin(clientId, (ClientRepresentation clientRep) -> {
+                    clientRep.setEnabled(false);
+                });
+                fail();
+            } catch (ClientPolicyException cpe) {
+                assertEquals(ClientPolicyEvent.UPDATED.toString(), cpe.getError());
+            }
+            assertEquals(false, getClientByAdmin(clientId).isEnabled());
+
+            updateExecutor(executorName, (ComponentRepresentation provider) -> {
+                provider.getConfig().put(TestRaiseExeptionExecutorFactory.TARGET_CP_EVENTS, new ArrayList<>(Arrays.asList(ClientPolicyEvent.VIEW.toString())));
+            });
+            try {
+                getClientByAdmin(clientId);
+            } catch (ClientPolicyException cpe) {
+                assertEquals(ClientPolicyEvent.VIEW.toString(), cpe.getError());
+            }
+
+            updateExecutor(executorName, (ComponentRepresentation provider) -> {
+                provider.getConfig().put(TestRaiseExeptionExecutorFactory.TARGET_CP_EVENTS, new ArrayList<>(Arrays.asList(ClientPolicyEvent.UNREGISTER.toString())));
+            });
+            try {
+                deleteClientByAdmin(clientId);
+            } catch (ClientPolicyException cpe) {
+                assertEquals(ClientPolicyEvent.UNREGISTER.toString(), cpe.getError());
+            }
+        } finally {
+            updateExecutor(executorName, (ComponentRepresentation provider) -> {
+                provider.getConfig().put(TestRaiseExeptionExecutorFactory.TARGET_CP_EVENTS, new ArrayList<>());
```

Comment:
new ArrayList<>()); 
could be replaced with immutable Collections.singletonList();

Reply:
@valb3r Thank you. `Collections.singletonList((String)null)` might be ok.